src/modules/package.rs

```diff
@@ -78,6 +76,19 @@ fn get_poetry_version(context: &Context, config: &PackageConfig) -> Option<Strin
     format_version(raw_version, config.version_format)
 }
 
+fn get_pep621_version(pyproject: &toml::Value, config: &PackageConfig) -> Option<String> {
+    let raw_version = pyproject.get("project")?.get("version")?.as_str()?;
+    format_version(raw_version, config.version_format)
+}
+
+fn get_pyproject_version(context: &Context, config: &PackageConfig) -> Option<String> {
+    let file_contents = context.read_file_from_pwd("pyproject.toml")?;
+    let pyproject_toml: toml::Value = toml::from_str(&file_contents).ok()?;
+
+    get_pep621_version(&pyproject_toml, config)
+        .or_else(|| get_poetry_version(&pyproject_toml, config))
+}
```

Comment:
This looks good, but I would prefer handling `format_version` in `get_pyproject_version`:
```rust
fn get_poetry_version<'a>(pyproject: &'a toml::Value) -> Option<&'a str> {
    pyproject
        .get("tool")?
        .get("poetry")?
        .get("version")?
        .as_str()
}

fn get_pep621_version<'a>(pyproject: &'a toml::Value) -> Option<&'a str> {
    pyproject.get("project")?.get("version")?.as_str()
}

fn get_pyproject_version(context: &Context, config: &PackageConfig) -> Option<String> {
    let file_contents = context.read_file_from_pwd("pyproject.toml")?;
    let pyproject_toml: toml::Value = toml::from_str(&file_contents).ok()?;

    get_pep621_version(&pyproject_toml)
        .or_else(|| get_poetry_version(&pyproject_toml))
        .and_then(|raw_version| format_version(raw_version, config.version_format))
}
```

Reply:
Done 👍🏻 When I implemented your snippet with the explicit lifetime annotations I got a clippy warning saying that could be elided so I just did it without them